experimental/traffic_router_golang/go.mod

```diff
@@ -0,0 +1,21 @@
+module traffic_router_golang
```

Comment:
There is already a [`go.mod`](https://github.com/apache/trafficcontrol/blob/master/go.mod) file in the top-level directory. It's convenient to track Go dependencies for the whole project in one place, and if the experimental traffic_router_golang package had its own `go.mod`, the top-level `go.mod` would not track its dependencies.

Reply:
> There is already a [`go.mod`](https://github.com/apache/trafficcontrol/blob/master/go.mod) file in the top-level directory. It's convenient to track Go dependencies for the whole project in one place, and if the experimental traffic_router_golang package had its own `go.mod`, the top-level `go.mod` would not track its dependencies.

Do you know how to compile package level for example to compile traffic_router_golang , I tied using the top level go.mod to build the package, but failed to do.

Reply:
Using Go 1.17.4,
```
cd experimental/traffic_router_golang
go mod download
go build
```

Reply:
Thanks, but got some errors, fixed it by running `go mod vendor` after running `go mod download`

Reply:
Ah right, `go mod vendor` is necessary since we vendor most dependencies already.